### PR TITLE
Kafka archiver completed with python libraries included

### PR DESCRIPTION
**Pull request by abigailrose**
Also includes some PDK documentation that was mistakenly done on this branch (primarily just README updates for the archiver testing process).

### CODE REVIEW

File: pscheduler-archiver-kafka/kafka/kafka.txt [hunk 1]
```diff
@@ -1 +1,5 @@
-DOCUMENTATION
+DOCUMENTATION FOR THE APACHE KAFKA ARCHIVER
+
+This archiver will produce an Apache Kafka message with a topic specified in the input JSON and send it to the cluster indicated by the server addresses given in the input JSON.
```
Comment (mfeit-internet2) [inline pscheduler-archiver-kafka/kafka/kafka.txt]: Plain text files should be wrapped.

File: pscheduler-archiver-kafka/kafka/archive [hunk 2]
```diff
@@ -14,14 +14,13 @@ def archive(json):
...
     try:
         producer = KafkaProducer(bootstrap_servers=[data["server-address"]])
         producer.send(data["topic"], pscheduler.json_dump(json["result"])).get(timeout=30)
         
     except Exception as e:
-        print(str(e))
+        #print(str(e))
 	result = {
             "succeeded": False,
             "error": "Failed to send message"
```
Comment (mfeit-internet2) [inline pscheduler-archiver-kafka/kafka/archive]: This should include what the exception was so the end user has some idea why it failed.

File: pscheduler-archiver-kafka/kafka/data-is-valid [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        "valid": False,
+        "error": str(ex)
+        })
+
+data_validator = {
+    #Data validation json object
+    "type": "object",
+    "properties": {
+        "topic": {"$ref": "#/pScheduler/String"},
+        "server-address": {"$ref": "#/pScheduler/String"}
```
Comment (mfeit-internet2) [inline pscheduler-archiver-kafka/kafka/data-is-valid]: This should be shortened to just `server`. 

Also take a look at what other parameters are available in `KafkaProducer` that might be worth passing allowing the user to pass through.  The client identifier and acks look useful.  At some point in the future, we'll probably have to get into authentication, but not this round.

File: pscheduler-archiver-kafka/kafka/enumerate [hunk 4]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sed -e 1d
+{
+    "schema": 1,
+    "name": "kafka",
+    "description": "EDIT ME",
```
Comment (mfeit-internet2) [inline pscheduler-archiver-kafka/kafka/enumerate]: Need a description here.

File: pscheduler-archiver-kafka/kafka/archive (hunk 2 above)
Comment (abigailrose) [reply pscheduler-archiver-kafka/kafka/archive]: So add it to the error part of the JSON, something like appending "due to error" +  str(e)?

Comment (mfeit-internet2) [reply pscheduler-archiver-kafka/kafka/archive]: The Pythonic way would be `"Failed to send message: %s" % (str(e))`

**Pull request merged**